v2/spiffetls/tlsconfig/config_test.go

```diff
@@ -0,0 +1,687 @@
+	assert.Nil(t, config.NameToCertificate)
+	assert.NotNil(t, config.VerifyPeerCertificate)
+}
+
+func TestMTLSWebServerConfig(t *testing.T) {
+	// Create trust domain, bundle and tls certificate
+	trustDomain := spiffeid.RequireTrustDomainFromString("test.domain")
+	bundle := x509bundle.New(trustDomain)
+	tlsCert := &tls.Certificate{Certificate: [][]byte{[]byte("body")}}
+
+	// Call TLSClientConfig
+	config := tlsconfig.MTLSWebServerConfig(tlsCert, bundle, tlsconfig.AuthorizeAny())
+
+	// Expected AuthFields
+	assert.Equal(t, []tls.Certificate{*tlsCert}, config.Certificates)
+	assert.Equal(t, tls.RequireAndVerifyClientCert, config.ClientAuth)
+	assert.Nil(t, config.GetCertificate)
+	assert.Nil(t, config.GetClientCertificate)
+	assert.True(t, config.InsecureSkipVerify)
+	assert.Nil(t, config.NameToCertificate)
+	assert.NotNil(t, config.VerifyPeerCertificate)
+}
+
+func TestHookMTLSWebServerConfig(t *testing.T) {
+	// Create trust domain, bundle and tls certificate
+	trustDomain := spiffeid.RequireTrustDomainFromString("test.domain")
+	bundle := x509bundle.New(trustDomain)
+	tlsCert := &tls.Certificate{Certificate: [][]byte{[]byte("body")}}
+
+	// Create test config
+	config := createTestTlsConfig()
+
+	// Call HookTLSClientConfig
+	tlsconfig.HookMTLSWebServerConfig(config, tlsCert, bundle, tlsconfig.AuthorizeAny())
+
+	// Expected AuthFields
+	assert.Equal(t, []tls.Certificate{*tlsCert}, config.Certificates)
+	assert.Equal(t, tls.RequireAndVerifyClientCert, config.ClientAuth)
+	assert.Nil(t, config.GetCertificate)
+	assert.Nil(t, config.GetClientCertificate)
+	assert.True(t, config.InsecureSkipVerify)
+	assert.Nil(t, config.NameToCertificate)
+	assert.NotNil(t, config.VerifyPeerCertificate)
+}
+
+func TestGetCertificate(t *testing.T) {
+	testCases := []struct {
+		name          string
+		source        *fakeSource
+		err           string
+		expectedCerts [][]byte
+	}{
+		{
+			name: "success",
+			source: &fakeSource{
+				err: nil,
+				svid: &x509svid.SVID{
+					ID: spiffeid.Must("test.domain", "host"),
+					Certificates: []*x509.Certificate{
+						{Raw: []byte("body")},
+					},
+				},
+			},
+			expectedCerts: [][]byte{[]byte("body")},
+		},
+		{
+			name: "source return error",
+			source: &fakeSource{
+				err: errors.New("some error"),
+			},
+			err: "some error",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			getCertificate := tlsconfig.GetCertificate(testCase.source)
+			require.NotNil(t, getCertificate)
+
+			//  Execute getCertificate
+			tlsCert, err := getCertificate(&tls.ClientHelloInfo{})
+			if testCase.err != "" {
+				require.EqualError(t, err, testCase.err)
+				require.Nil(t, tlsCert)
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, testCase.expectedCerts, tlsCert.Certificate)
+		})
+	}
+}
+
+func createTestTlsConfig() *tls.Config {
+	tlsCert := tls.Certificate{Certificate: [][]byte{[]byte("body")}}
+	return &tls.Config{
```

Comment:
This config should probably set all of the fields in the config so we can assert that the hook functions don't modify fields that aren't required to accomplish SPIFFE authentication.